flake-modules/dev/new-plugin.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import os
+import re
+from argparse import ArgumentParser
+
+# Template for default.nix
+default_nix_template = """{{ lib, ... }}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {{
+  name = "{name}";
+  originalName = "{originalName}";
+  package = "{package}";
+
+  maintainers = [ lib.maintainers.YOUR_NAME ];
+
+  settingsOptions = {{
+    exampleOption = lib.nixvim.defaultNullOpts.mkBool false ''
+      Example option for the {name} plugin.
+    '';
+  }};
+}}
+"""
+
+# Template for test file
+test_nix_template = """{{
+  empty = {{
+    plugins.{name}.enable = true;
+  }};
+}}
+"""
+
+
+def kebab_case(input_string):
+    # Convert to snake_case and remove trailing underscores
+    return (
+        re.sub(r"[\W_]+", "-", input_string)
+        .lower()
+        .replace("nvim", "")
+        .rstrip("-")
+        .lstrip("-")
+    )
+
+
+def create_nix_file(file_path, template, name, originalName, package):
+    content = template.format(name=name, originalName=originalName, package=package)
+    with open(file_path, "w") as f:
+        f.write(content)
```

Comment:
I'm thinking we should extract creating dirs + writing text to a file to a helper fn.

That way the `main()` fn doesn't have to take responsibility for ensuring dirs exist and without having to make these `create_*_file` fns more repetitive.

```py
def write_to_file(file_path, content: str):
    os.makedirs(os.path.dirname(file_path), exist_ok=True)
    with open(file_path, "w") as f:
        f.write(content)
```